test/cases/cloud-provider-config.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+KUBELET_UNIT_DIR="/etc/systemd/system/kubelet.service.d"
+
+function fail() {
+  echo "❌ Test Failed:" "$@"
+  find $KUBELET_UNIT_DIR -type f | xargs cat
+  exit 1
+}
+
+echo "--> Should use in-tree cloud provider when below k8s version 1.26"
+# This variable is used to override the default value in the kubelet mock
+export KUBELET_VERSION=v1.25.5-eks-ba74326
+EXIT_CODE=0
+/etc/eks/bootstrap.sh \
+  --b64-cluster-ca dGVzdA== \
+  --apiserver-endpoint http://my-api-endpoint \
+  test || EXIT_CODE=$?
+if [[ ${EXIT_CODE} -ne 0 ]]; then
+  fail "expected a zero exit code but got '${EXIT_CODE}'"
+fi
+EXIT_CODE=0
```

Comment:
Doesn't look like this is necessary because we'll fail otherwise.

Reply:
Fair enough, I just put this before every test case out of habit.